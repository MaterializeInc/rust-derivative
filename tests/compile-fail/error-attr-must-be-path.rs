#[cfg(feature = "use_core")]
extern crate core;

#[macro_use]
extern crate derivative;

#[derive(Derivative)]
#[derivative(Clone)]
struct Err1 {
    ok: u8,
    #[derivative(Clone(clone_with=42))]
    err1: u8,
}

#[derive(Derivative)]
#[derivative(Debug)]
struct Err2 {
    ok: u8,
    #[derivative(Debug(format_with=42))]
    err2: u8,
}

#[derive(Derivative)]
#[derivative(PartialEq)]
struct Err3 {
    ok: u8,
    #[derivative(PartialEq(compare_with=42))]
    err3: u8,
}

#[derive(Derivative)]
#[derivative(Default)]
struct Err4 {
    ok: u8,
    #[derivative(Default(value=42))]
    err4: u8,
}

#[derive(Derivative)]
#[derivative(Hash)]
struct Err5 {
    ok: u8,
    #[derivative(PartialEq(hash_with=42))]
    err5: u8,
}

fn main() {}