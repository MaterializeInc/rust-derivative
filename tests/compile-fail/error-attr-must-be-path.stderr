error: expected derivative clone_with attribute to be a string: `clone_with = "..."`
  --> $DIR/error-attr-must-be-path.rs:11:35
   |
11 |     #[derivative(Clone(clone_with=42))]
   |                                   ^^

error: expected derivative format_with attribute to be a string: `format_with = "..."`
  --> $DIR/error-attr-must-be-path.rs:19:36
   |
19 |     #[derivative(Debug(format_with=42))]
   |                                    ^^

error: expected derivative compare_with attribute to be a string: `compare_with = "..."`
  --> $DIR/error-attr-must-be-path.rs:27:41
   |
27 |     #[derivative(PartialEq(compare_with=42))]
   |                                         ^^

error: expected derivative value attribute to be a string: `value = "..."`
  --> $DIR/error-attr-must-be-path.rs:35:32
   |
35 |     #[derivative(Default(value=42))]
   |                                ^^

error: expected derivative hash_with attribute to be a string: `hash_with = "..."`
  --> $DIR/error-attr-must-be-path.rs:43:38
   |
43 |     #[derivative(PartialEq(hash_with=42))]
   |                                      ^^
